stories/components/accordion.stories.tsx

```diff
@@ -337,16 +340,91 @@ export const UnmountChildren = {
 export function UnmountSomeChildren() {
   return (
     <Accordion>
-      <Accordion.Item title="Always render children" defaultOpened>
+      <Accordion.Item title="Always render children" defaultOpen>
         <Count />
       </Accordion.Item>
-      <Accordion.Item title="Unmount children" defaultOpened unmountChildren>
+      <Accordion.Item title="Unmount children" defaultOpen unmountChildren>
         <Count />
       </Accordion.Item>
     </Accordion>
   );
 }
 
+export function WithToolbar() {
+  return (
+    <Accordion>
+      <Accordion.Item
+        title="First"
+        toolbar={<Button icon="minus" variant="minimal" />}
+      >
+        First content
+      </Accordion.Item>
+      <Accordion.Item
+        title="Second"
+        defaultOpen
+        toolbar={<Button icon="minus" variant="minimal" />}
+      >
+        Second content
+      </Accordion.Item>
+    </Accordion>
+  );
+}
+
+type ControlledTitle = 'First' | 'Second';
+
+export function ControlledState() {
+  const [openItems, setOpenItems] = useState<ControlledTitle[]>(['First']);
+
+  const addItem = useCallback(
+    (title: ControlledTitle) => {
+      setOpenItems((openItems) => Array.from(new Set([...openItems, title])));
+    },
+    [setOpenItems],
+  );
+
+  const removeItem = useCallback(
+    (title: ControlledTitle) => {
+      setOpenItems((openItems) => openItems.filter((item) => item !== title));
+    },
+    [setOpenItems],
+  );
+
+  return (
+    <Accordion>
+      <Accordion.Item<ControlledTitle>
+        title="First"
+        open={openItems.includes('First')}
+        onOpenChange={(isOpen) =>
+          isOpen ? addItem('First') : removeItem('First')
+        }
+      >
+        First content
+      </Accordion.Item>
+      <Accordion.Item<ControlledTitle>
+        title="Second"
+        open={openItems.includes('Second')}
+        onOpenChange={(isOpen) =>
+          isOpen ? addItem('Second') : removeItem('Second')
+        }
+      >
+        Second content
+      </Accordion.Item>
+    </Accordion>
+  );
```

Comment:
I'm not convinced by the controlled state api. It is verbose and repetitive.

I think the state should be managed at `Accordion` level, with `opens` / `defaultOpens` / `onChange` props.

Reply:
The main use case is to use the "uncontrolled" version and use the `useAccordionToggle` hook if we need to control the opened / closed items from outside.

The `defaultOpens` prop on `Accordion` would make it more difficult to use it for this use case (need to repeat the `id` in 2 different components).

I agree that the story is verbose, but in a real case scenario of controlled state we would have a simple `dispatch` on change and a wrapper component around `Accordion.Item` to make things much less verbose.

Reply:
Yes, please provide a story which control the state with `useAccordionToggle`

Reply:
New story using a reducer + context to manage the external state provided.